src/transaction/rw_transaction.rs

```diff
@@ -152,22 +152,21 @@ impl<'db, 'txn> RwTransaction<'db> {
     ///     let rw = db.rw_transaction()?;
     ///
     ///     // Remove a value
-    ///     rw.remove(Data { id: 1 })?;
+    ///     let old_value = rw.remove(Data { id: 1 })?;
     ///
     ///     // /!\ Don't forget to commit the transaction
     ///     rw.commit()?;
     ///
     ///     Ok(())
     /// }
     /// ```
-    // TODO: Return the Option<T> of the removed value
-    pub fn remove<T: Input>(&self, item: T) -> Result<()> {
+    pub fn remove<T: Input>(&self, item: T) -> Result<T> {
         let (watcher_request, binary_value) = self
             .internal
             .concrete_remove(T::native_db_model(), item.to_item())?;
-        let event = Event::new_delete(binary_value);
+        let event = Event::new_delete(binary_value.clone());
         self.batch.borrow_mut().add(watcher_request, event);
-        Ok(())
```

Comment:
@AlvinKuruvilla you have to try to return a value of type `T` that comes from `binary_value` instead of `item`. This ensures that the value you return is actually the one from the database. To do this, you need to convert `binary_value` into type `T`. 

`binary_value` is a `DatabaseOutputValue`, so you'll use the [`inner`](https://docs.rs/native_db/latest/native_db/db_type/struct.DatabaseOutputValue.html#method.inner) method to convert it.

Please ask any questions you might have, even if there's a delay in my response. Sorry for any wait time.

Reply:
No apologies necessary thank you for taking the time to explain. I will keep it in mind for the future :)

Reply:
Thank you for the help squashing the commits; that was my first time doing that. I hope that fixes the CI issues.